src/scripts/main.js

```diff
@@ -1,3 +1,50 @@
 'use strict';
 
-// write code here
+document.addEventListener('DOMContentLoaded', () => {
+  const table = document.querySelector('table');
+  const headers = table.querySelectorAll('th');
+  const tbody = table.querySelector('tbody');
+
+  headers.forEach((header, index) => {
+    header.addEventListener('click', () => {
+      sortTableByColumn(index);
+    });
+  });
+
+  function sortTableByColumn(columnIndex) {
+    const rows = Array.from(tbody.querySelectorAll('tr'));
+
+    rows.sort((a, b) => {
+      const cellA = a.children[columnIndex].textContent.trim();
+      const cellB = b.children[columnIndex].textContent.trim();
+
+      const valueA = parseValue(cellA);
+      const valueB = parseValue(cellB);
+
+      if (valueA < valueB) {
+        return -1;
+      }
+
+      if (valueA > valueB) {
+        return 1;
+      }
+
+      return 0;
+    });
+
+    tbody.innerHTML = '';
+    rows.forEach((row) => tbody.appendChild(row));
+  }
+
+  function parseValue(value) {
+    if (value.includes('$')) {
+      return parseFloat(value.replace(/[$,]/g, ''));
+    }
+
+    if (!isNaN(value)) {
```

Comment:
The check `!isNaN(value)` might not work as expected for all cases, especially if `value` is an empty string or contains non-numeric characters. Consider using `Number.isNaN()` for a more reliable check.